Userland/Libraries/LibCore/Socket.cpp

```diff
@@ -200,14 +215,22 @@ void PosixSocketHelper::setup_notifier()
 ErrorOr<NonnullOwnPtr<TCPSocket>> TCPSocket::connect(ByteString const& host, u16 port)
 {
     auto ip_address = TRY(resolve_host(host, SocketType::Stream));
-    return connect(SocketAddress { ip_address, port });
+
+    return ip_address.visit(
+            [port](IPv6Address ipv6_address) { return connect(SocketAddress { ipv6_address, port }); },
+            [port](IPv4Address ipv4_address) { return connect(SocketAddress { ipv4_address, port }); }
+    );
 }
 
 ErrorOr<NonnullOwnPtr<TCPSocket>> TCPSocket::connect(SocketAddress const& address)
 {
+    auto socketDomain = SocketDomain::Inet6;
     auto socket = TRY(adopt_nonnull_own_or_enomem(new (nothrow) TCPSocket()));
 
-    auto fd = TRY(create_fd(SocketDomain::Inet, SocketType::Stream));
+    if(address.type() == SocketAddress::Type::IPv4)
+        socketDomain = SocketDomain::Inet;
```

Comment:
Not 100% like this pattern,
but at least move the variable next to the if statement, and use a snake_case name

Reply:
Not liking as in this will lead to funny issues when another socket domain or address type is added